### PR TITLE
First pass refactoring scrapers into separate modules

### LINKED ISSUE #95: modularise and break up code
- [x] Break up spoiler.py into independent modules (#98)
- [ ] Make sure each module has no weird cross dependencies
- [ ] Making sure each modules state is properly isolated and that modules pass consistent values out.
- [ ] Writing tests for modules to make sure that when we make changes, that we get the expected result on the other side. :)

Taken from: https://github.com/Cockatrice/Magic-Spoiler/pull/98#issuecomment-313128570

---

*old issue text:*

>- Break spoiler.py up into a few modules (mtgs/scryfall/etc)
>- Better manageable, read and maintain
>
><br>
>
>Also consider separate "mtgs_transformations/corrections.json" (https://github.com/Cockatrice/Magic-Spoiler/blob/4e43b90156b5ff31ff273ba8afb5aeef3859f6eb/spoilers.py#L88-L104)
>- More robust core application
>- Split essentials (process modifications) from actual data modifications 
>- Easier to contribute for non-programmers
>
>Basic workflow could be:
>Get+scrape data (one for each service?) -> correct errors -> bake cards -> build release files
>
>Conversation regarding that on Gitter:  :point_up: [July 3, 2017 3:51 PM](https://gitter.im/Cockatrice/Magic-Spoiler?at=595a4bc8329651f46e4adf15)

### PR DESCRIPTION
**Pull request by leematos**
This _could_ close #95. It needs a ton more work to really untangle to make sure the modules don't have weird cross dependencies but as a first pass it makes things a little bit more organized 🎉 I ran it locally and didn't hit any errors so I think it's safe to merge.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -30,94 +37,114 @@ def load_json(json_file, lib_to_use):
...
+        mtgs = spoilers.correct_cards(
+            mtgs, manual_sets[setinfo['setname']]['cards'], card_corrections, delete_cards)  # fix using the fixfiles
+        mtgjson = spoilers.get_image_urls(
+            mtgs, presets['isfullspoil'], setinfo['setname'], setinfo['setlongname'], setinfo['setsize'], setinfo)  # get images
         if presets['scryfallComparison']:
-            scryfall = spoilers.get_scryfall(
-                'https://api.scryfall.com/cards/search?q=++e:' + setinfo['setname'].lower())
-            mtgjson = spoilers.smash_mtgs_scryfall(mtgs, scryfall)
+            scryfall = scryfall_scraper.get_scryfall(
+                'https://api.scryfall_scraper.com/cards/search?q=++e:' + setinfo['setname'].lower())
```
Comment (leematos) [inline main.py]: This needs to get fixed.

**Review by dev-id — CHANGES_REQUESTED** [review]
Let's remove all unneeded imports

**Comment by tooomm** [comment]
Can you explain a bit more what you did so far, please @leematos ?

If I checked correctly you basically just split up
`spoiler.py` --> in separate `*_scrapers.py` modules for each source (mtgs, scryfall, mythic and wizards)
as of now?

Because of all that format and space changes I might got lost on the way. :D

Any other code changes or refactors so far that I might missed?

**Review by dev-id — APPROVED** [review]

**Comment by leematos** [comment]
@tooomm Yep, for now I just separated them out into separate files (modules) so we can start to work to make them self-contained and make sure the flow of the program is a little bit clearer. A module should "do one thing" (which can turn into a semantic nightmare) but it seems to me that `spoilers.py` was defining a couple of different scrapers which is why I split them up here. 

The next steps after this would be:

1. Make sure each module has no weird cross dependencies.
1. Making sure each modules state is properly isolated and that modules pass consistent values out.
1. Writing tests for modules to make sure that when we make changes, that we get the expected result on the other side. :)

**Comment by dev-id** [comment]
Can you see about matching the previous spacing conventions?

Edit: I guess there is no real convention or standard. I think my IDE will respect what's present...

**Comment by leematos** [comment]
I was actually worried about that @tritoch. I ran it though a PEP8 linter and just YOLO'd it since we should be following PEP8 anyway https://www.python.org/dev/peps/pep-0008/

**Comment by dev-id** [comment]
Look at this guy, using style guides.

**Comment by tooomm** [comment]
put that into contributing? (if that's what we like)

**Comment by leematos** [comment]
I just squashed those fix import commits to clean up the history. This one's ready to merge! <3

**Pull request merged**